tests/rust/run_tests.sh

```diff
@@ -2,8 +2,19 @@
 
 set -e
 
-# Treat warnings as errors
-export RUSTFLAGS="-D warnings"
+
+# Let's debug RUSTFLAGS
+echo "RUSTFLAGS=$RUSTFLAGS"
+export WITH_FATAL_WARNINGS="no"
+echo "WITH_FATAL_WARNINGS=$WITH_FATAL_WARNING"
```

Comment:
If you want to disable warnings, you should `WITH_FATAL_WARNINGS` to `no` here:

https://github.com/cossacklabs/themis/blob/00f9824b3c8d5c3e40737158851e35967ebc6703/.github/workflows/test-rust.yaml#L24-L25

I think if you do it like that, you can even keep `stable` Rust in build rotation. You can even try allowing warnings _only_ for stable but keep them fatal when building with 1.56.

Reply:
> You can even try allowing warnings only for stable but keep them fatal when building with 1.56

shouldn't we do this in master? and just pin 1.56 in stable?

Reply:
There is merit in testing the maintenance release with modern stable Rust – we'll get a warning if there is Bad Code detectable only by newer compiler, as that might be something we want to fix.

There is merit in having at least some builds treat warnings as errors, so that changes that add new accidental warnings fail the build.

Current release/0.14 is _not_ warning-free with stable Rust but is with Rust 1.56.

Considering all that, there are several options. Two easy ones:

* `WITH_FATAL_WARNINGS: no` — don't think too much about warnings
* pin Rust to 1.56 — then effectively no warnings are generated

I think we spend too much time bikeshedding this, so I'd rather pin Rust to 1.56 here and leave fatal warnings on. And on master run both stable and 1.58 and still fatal warnings on.

In any case, the hunk in `run_tests.sh` should be removed 😄

Reply:
just pin 1.56 in stable!

Reply:
I will remake the PR. Now I converted this into a draft again. I appreciate your help.